highfidelity/hifi
# [Case 7049]: Remove unused fields and checkboxes from Edit.js

* Changes:
    * Removed duplicate color fields for shapes.
    * Remove basic color section from non-shape entity Properties tab.
    * Fixes superfluous fields issue with Particle Effects.
        * Previously all entity type sections were visible in Properties tab when a particle effect was the current selection.
    * Gives collision section a header for consistency.
    * Various lint fixes.

Test Plan:
(Note:  Sections of the Properties Tab are listed in the order they're expected to be listed.)

* Create Model Entity
    * Test Model Link (if needed): https://s3-us-west-1.amazonaws.com/hifi-content/jazmin/production/assets/robot.fbx
    * Properties Tab should have the following areas present
        * General Section (Selection Type, Locked Status, Visible Status, Name)
        * Model Section
        * Collision Section
        * Physical Section
        * Spatial Section
        * Behavior Section
        * Hyperlink Section
* Create Cube or Sphere
    * Properties Tab should have the following areas present
        * General Section (Selection Type, Locked Status, Visible Status)
        * Shape Section (Shape Type dropdown list, Name Field, & Entity Color)
        * Collision Section
        * Physical Section
        * Spatial Section
        * Behavior Section
        * Hyperlink Section
* Create Non-Cube/Sphere shape (ex: Octagon)
    * Properties Tab should have the following areas present
        * General Section (Selection Type, Locked Status, Visible Status)
        * Shape Section (Shape Type dropdown list, Name Field, & Entity Color)
        * Collision Section
        * Physical Section
        * Spatial Section
        * Behavior Section
        * Hyperlink Section
* Create Light Entity
    * Properties Tab should have the following areas present
        * General Section (Selection Type, Locked Status, Visible Status, Name)
        * Light Section
        * Physical Section
        * Spatial Section
        * Behavior Section
        * Collision Section
        * Hyperlink Section
* Create Text Entity
    * Properties Tab should have the following areas present
        * General Section (Selection Type, Locked Status, Visible Status, Name)
        * Text Section
        * Collision Section
        * Physical Section
        * Spatial Section
        * Behavior Section
        * Hyperlink Section
* Create Web Entity
    * Properties Tab should have the following areas present
        * General Section (Selection Type, Locked Status, Visible Status, Name)
        * Web Section
        * Collision Section
        * Physical Section
        * Spatial Section
        * Behavior Section
        * Hyperlink Section
* Create Zone Entity
    * Properties Tab should have the following areas present
        * General Section (Selection Type, Locked Status, Visible Status, Name)
        * Zone Section
        * Physical Section
        * Spatial Section
        * Behavior Section
        * Collision Section
        * Hyperlink Section
* Create ParticleEffect Entity
    * User should be directed to Particle Effect Tab (nothing has changed here)
    * Select the Properties Tab
        * Properties Tab should have the following areas present
            * General Section (Selection Type, Locked Status, Visible Status, Name)
            * Collision Section
            * Physical Section
            * Spatial Section
            * Behavior Section
